lib/ddtrace/transport/trace_formatter.rb

```diff
@@ -47,6 +47,7 @@ def format!
         tag_rate_limiter_rate!
         tag_sample_rate!
         tag_sampling_decision_maker!
+        tag_high_order_trace_id!
```

Comment:
We should also adjust the the `span.trace_id` on all spans in the `TraceFormatter`, so the `TraceFormatter` always makes sure the `TraceSegment` provided is ready to be serialized without any changes.

This simplifies the serializer's logic.

Reply:
I actually hesitate to commit to this change since 

`TraceFormatter` is performing at the trace level attributes which is a constraint of agent's API. I would prefer the current approach of decorating `SerailizableSpan` than mutating `Span` from `TraceFormatter`.